File: templates/python/src/trading_game_args.py
from dataclasses import dataclass

from typing import Tuple, List


# R=10 MM=50 MQ=0 OM=50 OQ=0
# R=4 MM=23 MQ=2 OM=15 OQ=4 10/5 15/16 3/14
@dataclass(frozen=True)
class TradingGameArgs:
    remaining_items: int
    my_money: int
    my_quantity: int
    other_money: int
    other_quantity: int
    bids_history: List[Tuple[int, int]]

File: templates/java/src/main/java/de/finatix/TradingGameArgs.java
package de.finatix;

import org.apache.commons.lang3.tuple.Pair;

import java.util.List;

public record TradingGameArgs(
        int remainingItems,
        int myMoney,
        int myQuantity,
        int otherMoney,
        int otherQuantity,
        List<Pair<Integer, Integer>> bidsHistory
) {
}
